### DISCUSSION TITLE
When enabled, all cameras stuck in "Live View" mode

### DISCUSSION THREAD

**Discussion by rtech73** [tsightler/ring-mqtt#652] [Support Q&A]
Hello all!   

I just did a reinstall of Ring-MQTT (I had a few cameras disappear, and it seemed the ol' "IT Crowd" solution was best).   Upon reinstallation, everything was back and working like it should - except.   All of my cameras are now stuck in "live view" mode, and my battery cameras only last 24 hours.  Checking the "event history" on the camera shows a constant stream of 10 minute "live views" (which is what is draining the batteries).   I can't seem to figure out what might be causing this, or how to fix it.   Using it as an add-on in Home Assistant (I think that's assumed, but just in case!).   When I disable the addon, the cameras go back to normal.   Any suggestions as to what I might look for?   Again, this was basically a fresh install!   

Thanks so much!

**Comment by jeffgraham-hub** [comment]
Msybe this will help

From the FAQ
 Q) Streams keep starting all the time even when I'm not viewing anything
A) In Home Assistant, do not use the "Preload Stream" option and make sure the Camera View setting in the Picture Glance card is set to "auto" instead of "live". If you use the "live" setting, the card will attempt to start streams in the background for faster startup when you bring up the card. This is fine for local cameras but, because Ring cameras do not send motion events during streams, having streams running all the time will cause motion events to be missed and, since all streaming goes through Ring servers on the Internet, you will use a lot of bandwidth as well.

There have also been reports of certain components connecting to live streams with when no password is set on the RTSP server, I've not personally seen this behavior but multiple users have sent log examples showing "something" connecting to the RTSP server which triggers the stream to start. The simplest way to solve this is to set a livestream username/password so that services which automatically discover RTSP endpoints cannot connect and start streams.

**Comment by hijamie32** [comment]
Just wanted to say I have experienced exactly the same issue, battery drain and video clips every ten minutes!

I found simply logging back in on the ring addon web page cleared the problem for a few days - but came back last night (had been working flawlessly for a few months prior).

I have had a few of the settings above on since installing and not had a problem till the last couple of weeks or so....

Cheers

**Comment by tsightler** [comment]
I'll just answer with the same thing I always do, if streams are starting, then some client is requesting them to start.  You have to determine what client is connecting to the stream and stop it.  Simplest solution is to set a username/password for the livestream and see if everything stops, then work your way through the configured clients and carefully verify all settings are correct (no preload stream, no "live" settings, snapshot URL configured correctly, etc).

If you change the livestream username/password and streams still start in that case, then open an issue and provide logs but, otherwise, there's not much I can do for you except offer this troubleshooting advice.  Over the years since streaming was added I've worked through dozens of reports like this and in every single case, except one, there's been some client misconfiguration.  In the one case there was a bug in go2rtc shortly after I switched from rtsp-simple-server to go2rtc.

**Reply by hijamie32** [reply]
Thanks for the reply.

Was more intrigued by the fact someone else seems to be having the same problem as me (although there's is after a reinstall).

May not be anything directly related to this add-on, but just strange it randomly started happening after not changing anything so hopefully be related and may be the same issue as the other person.

I have already reauthenticated so takes a while for the problem to start again - once it does I will enable username + password and also disable the WebRTC Addon a bit later after if the issue isn't resolved and will post back here.

Cheers again for the great addon :)

**Comment by tsightler** [comment]
I'll add a few more troubleshooting hints here:

1) The ring-mqtt logs are your friend.  They will show you what is happening clearly.  Here's a sample of a working live stream from startup to shutdown:
```
2023-06-26T13:18:05.767Z ring-rtsp [go2rtc] DBG [rtsp] new consumer stream=3045113e46e4_live
2023-06-26T13:18:05.768Z ring-rtsp [go2rtc] DBG [exec] run url="exec:/app/ring-mqtt/scripts/start-stream.sh 3045113e46e4 live ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4 {output}"
2023-06-26T13:18:05.801Z ring-rtsp [Front Garage Side] Sending command to activate live stream ON-DEMAND
2023-06-26T13:18:05.804Z ring-mqtt [Front Garage Side] Received set live stream state ON-DEMAND rtsp://127.0.0.1:8554/d206fe7b7bc39a6e09c89670e5c5db06
2023-06-26T13:18:05.804Z ring-mqtt [Front Garage Side] ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4/stream/state ON
2023-06-26T13:18:05.805Z ring-attr [Front Garage Side] ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4/stream/attributes {"status":"activating"}
2023-06-26T13:18:05.806Z ring-mqtt [Front Garage Side] Initializing a live stream session for Ring cloud
2023-06-26T13:18:05.903Z ring-rtsp [Front Garage Side] State indicates live stream is activating
2023-06-26T13:18:05.960Z ring-mqtt [Front Garage Side] Live stream session successfully initialized, starting worker
2023-06-26T13:18:05.967Z ring-wrtc [Front Garage Side] Live stream WebRTC worker received start command
2023-06-26T13:18:06.029Z ring-wrtc [Front Garage Side] Live stream transcoding process is starting
2023-06-26T13:18:06.221Z ring-wrtc [Front Garage Side] Websocket signaling for Ring cloud connected successfully
2023-06-26T13:18:06.499Z ring-wrtc [Front Garage Side] Live stream transcoding process has started
2023-06-26T13:18:06.886Z ring-attr [Front Garage Side] ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4/stream/attributes {"status":"active"}
2023-06-26T13:18:06.888Z ring-wrtc [Front Garage Side] Live stream WebRTC session is connected
2023-06-26T13:18:06.978Z ring-rtsp [Front Garage Side] State indicates live stream is active
2023-06-26T13:18:07.753Z ring-rtsp [go2rtc] DBG [exec] run launch=1.987355707s
2023-06-26T13:18:07.755Z ring-rtsp [go2rtc] DBG [streams] start producer url=exec:/app/ring-mqtt/scripts/start-stream.sh 3045113e46e4 live ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4 {output}
2023-06-26T13:18:16.069Z ring-rtsp [go2rtc] DBG [rtsp] handle=EOF
2023-06-26T13:18:16.069Z ring-rtsp [go2rtc] DBG [streams] stop producer url=exec:/app/ring-mqtt/scripts/start-stream.sh 3045113e46e4 live ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4 {output}
2023-06-26T13:18:16.069Z ring-rtsp [go2rtc] DBG [rtsp] disconnect stream=3045113e46e4_live
2023-06-26T13:18:16.080Z ring-wrtc [Front Garage Side] Live stream WebRTC session has disconnected
2023-06-26T13:18:16.081Z ring-mqtt [Front Garage Side] ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4/stream/state OFF
2023-06-26T13:18:16.081Z ring-attr [Front Garage Side] ring/a9ce28cb-fafb-421c-4508-29d54f48cbd3/camera/3045113e46e4/stream/attributes {"status":"inactive"}
2023-06-26T13:18:16.148Z ring-rtsp [Front Garage Side] State indicates live stream has gone inactive
```

Notice the very first log line in ring-mqtt is from go2rtc showing a new RTSP consumer for the live stream.  This is what triggers the entire process.  Once the consumer connects, go2rtc runs the "start-stream.sh" script which uses MQTT to signal ring-mqtt to start a stream.  You can clearly see this on the second line of the log.

The 3rd line shows the script starting and sending the MQTT request to start an "ON-DEMAND" stream (i.e. a client has connected and requested the stream to start) and the 4th line shows ring-mqtt responding that it received this request and will start the stream to the random ID enpoint for go2rtc.

The 5th and 6th lines are ring-mqtt communicating the status, the "ON" indicates that the stream is starting (so that the stream switch in HA will reflect the status), while the attribute indicates that the stream is still activating (the stream is inititializing but the active streaming hasn't started yet).

Line 7 indicates that ring-mqtt is connecting to the Ring API and requesting a live streaming session.

Line 8 indicates that the start-streaming.sh script received the activating state from ring-mqtt (mostly just for debugging)

Line 9 indicates that the Ring API responded with a session and that data is being sent to the WebRTC worker thread for the camera

Line 10 indicates that the WebRTC worker thread received the session information from the main ring-mqtt thread

Line 11 indicates that the WebRTC worker thread has established a connection with the Ring streaming servers and is starting the ffmpeg process for forwarding the stream to RTSP.

Line 12 indicates that the websocket control channel was established between the WebRTC worker thread and Ring servers.  This is a control channel used for the streaming session to exchange SDP data and otherwise communicate required information to maintain the stream

Line 13 indicates that the FFMPEG process has started and is starting to receive data

Line 14 is ring-mqtt sending state that the stream is now active

Line 15 shows that the WebRTC session is fully connected and data is starting to flow

Line 16 indicates that the start-stream.sh script received the message that the stream is now active

Line 17 indicates that go2rtc is now receiving stream data and shows how long from the time the client connect until streaming started (in this case just under 2 seconds)

Line 18 indicates that the go2rtc recognizes that the start-stream.sh script executed successfully.

Line 19 indicates that the streaming client disconnected from the RTSP stream

Line 20 indicates that go2rtc recognizes that the client disconnected the stream and it stops the start-stream.sh script

Line 21 indcates that go2rtc removed the RTSP client from the list of streaming clients so that it would not re-trigger the stream to start again

Line 22 indicates that ring-mqtt is disconnecting the WebRTC session

Line 23 & 24 show that the stream was successfully stopped and state was updated

Line 25 shows that the start-stream.sh script received the inactive state and exited with no errors.

**Comment by tsightler** [comment]
Out of all of that, probably the most interesting is the very first line the "new consumer", if you see this line in the log then "something" is connecting to the RTSP stream so of course the stream will start.  You have to figure out what that is.

I have no idea why re-authenticating would have anything to do with this.  The streaming engine barely uses anything to do with authentication (a single call to get a streaming session) and that can only be triggered by a client connecting to the stream.

**Comment by hijamie32** [comment]
Thanks, when the issue starts back up again I will watch the logs to see what comes up - with that breakdown it makes it much easier to understand so thanks.

In terms of Authentication I am not really sure, but I think when I was messing around last week that a restart of the Addon didn't solve it, but reauthentication did.

Also just checked and I already have a user / password defined in the config page.

Its frustrating as can't do much more until it starts again.

**Comment by rtech73** [comment]
Appreciate that there is a discussion here.   Of course something is triggering the live stream.   It's tough, though, to determine what it might be when literally nothing has changed in our setups.    hijamie32 and I will keep an eye on this, and hopefully, since our issues seem to be exactly the same, one of us will figure out what it is.   Whatever is triggering a live stream, it's doing it every 10 minutes to every single camera we own.   I'm sorry, but that feels like a bug to me.   :-)

**Reply by tsightler** [reply]
Of course there is always a chance there is a bug in ring-mqtt or some surrounding component, but nothing much has changed with the streaming startup code in the last 6 months, other than bumping go2rtc versions.

Also, I did provide a super easy method to determine if ring-mqtt is somehow at fault, simply change the livestream username or password and restart the addon.  Since that will block all external clients from connecting and triggering the streams, you will be able to quickly determine if ring-mqtt is somehow triggering the stream on its own.  If it does, then indeed, it's a ring-mqtt bug and you should open a bug report and provide full logs.  If not, then you can work through the cameras one at a time and see if you can reproduce the issue.

If you want assistance to troubleshoot the issue, you can open a support issue, but be prepared to share full details and logs of your setup as that will be required to solve the issue as, so far in this discussion, no one has provided anywhere close to the level of information that would be required to solve such a problem, even if it is caused by a bug.

**Comment by rtech73** [comment]
Actually of my 15 or so cameras, there are a few that are NOT stuck in this live stream loop.    Trying to see what the difference is.   i'll report back!

**Comment by rtech73** [comment]
Why would a camera that is disabled (in the list of devices in MQTT, I selected a camera that is victim to the live stream loop (is what I'll call it) and completely disabled it.   It's still having the same thing happen - which doesn't make sense to me on the surface.   Still digging.

**Reply by tsightler** [reply]
Live streaming cameras are not MQTT devices so disabling a camera there has no impact on the live cameras.  Only snapshots cameras use MQTT.

**Comment by tsightler** [comment]
For anyone experiencing this issue, by any chance have you not configured the still image URL setting or perhaps configured it incorrectly?  This can trigger this behavior. in some cases because, if there is no still image URL, then HA will attempt to get the images from the stream, but that will keep the stream alive.

**Reply by rtech73** [reply]
I don't have still images for any of my cameras.  Never have.   I'll go take some pictures and see if that makes a difference!

(as a side note, are we all assuming that we have these cameras' live streams set up through the generic camera integraion as well?  That's the only way to actually view the live streams, right?)

**Reply by tsightler** [reply]
Well, there are many ways since they are just RTSP streams, but the common way for Home Assistant is the generic IP camera, which supports RTSP.  However, I recommend enhancing the experience with the RTSPtoWebRTC integration and go2rtc addon as this provides the ideal experience of fast startup, low stream latency (<1 second) and quick shutdown (usually 10 seconds), while the native HA stream engine uses HLS which has longer startup, higher latency, and shuts down only after 5 minutes of complete inactivity of the stream.

The video streaming documentation already mentions that not configuring the still image URL can lead to the stream starting, but I'll admit it's not very clear, perhaps I should re-write this section to be more clear no that.
